### PR TITLE
solved #558 #552

### LINKED ISSUE #552: A MILLION MURGHUL SPAWNED FOR FROSTWOLF
image

### LINKED ISSUE #557: Illidan: Cliff Runner and Glaive Warrior Share Hotkey (Q) in Hunter Proving Grounds

### LINKED ISSUE #558: kael forces got given to Dalaran
image

### PR DESCRIPTION
**Pull request by TracyTheBoy**
closes #558 
closes #552
Put QuelThalasQuestSetup before Dalarans
Removed actions from QuestSeaWitch trigger after triggering once and made it so only Thrall can trigger it

### CODE REVIEW

File: src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public sealed class QuestSeaWitch : QuestData
         .AddAction(() =>
         {
           var triggerUnit = GetTriggerUnit();
-          if (GetOwningPlayer(triggerUnit) != FrostwolfSetup.Frostwolf?.Player) return;
+          if (triggerUnit != LegendFrostwolf.LegendThrall.Unit) return;
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs:44]: I think the previous condition is correct - seems to reflect what's in the old GUI version.
image

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
I can't say I would've thought of that quest setup reordering solution but it does make sense!

File: src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs [hunk 2]
```diff
@@ -58,6 +58,7 @@ public sealed class QuestSeaWitch : QuestData
           {
             murgul.IssueOrder("attack", new Point(-3372, -2261));
           }
+          TriggerClearActions(_rescueTrigger);
```
[the remark this answers is not in the document]
Comment (TracyTheBoy) [reply src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs:61]: Yea for some reason I only tried `_rescueTrigger.Destroy()` but that didn't exist

File: src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs (hunk 1 above)
Comment (TracyTheBoy) [reply src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs:44]: I just thought it'd make more sense thematically for the leader of the horde to recruit the units. We can revert if you want

**Comment by TracyTheBoy** [comment]
> I can't say I would've thought of that quest setup reordering solution but it does make sense!

Wierdly enough it does work the other way around in the vJass version.

File: src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs (hunk 1 above)
Comment (YakaryBovine) [reply src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs:44]: Yeah, I think we should revert and go with the old condition. If we use this new one we'd have to convey it to the player somehow. After all training Thrall in this scenario is not even mandatory.

File: src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (YakaryBovine) [reply src/WarcraftLegacies.Source/Quests/Frostwolf/QuestSeaWitch.cs]: Feel free to add it in `TriggerExtensions.cs`. If you don't I will at some point.

**Pull request merged**